sccp/figures/figure5.py

```diff
@@ -0,0 +1,72 @@
+"""
+Parafac2 implementation on PBMCs treated across treatments and patients
+"""
+import numpy as np
+from .common import subplotLabel, getSetup, plotSCCP_factors, renamePlotsCoH
+from ..imports.CoH import CoH_xarray
+from ..parafac2 import parafac2_nd
+from ..decomposition import plotR2X_CC
+
+
+def makeFigure():
+    """Get a list of the axis objects and create a figure."""
+    # Get list of axis objects
+    ax, f = getSetup((10, 20), (4, 3))
+
+    # Add subplot labels
+    subplotLabel(ax)
+
+    # Import of single cells: [Patient, Treatment, Cell, Marker]
+    cohXA, celltypeXA = CoH_xarray(saveXA=False)
+
+    # Shrink dataset
+    cohXA = cohXA.loc[:, :, :500, :]
+    celltypeXA = celltypeXA.loc[:, :, :500]
+
+    # Normalize here
+    cohXA.values /= np.nanmean(cohXA.values, axis=0)
```

Comment:
I'm not sure if this is the right normalization, but you should just perform the normalization here. This is a very fast operation, and it means you don't have to rebuild the data every time this changes.